lib/arc/file.ex

```diff
@@ -33,6 +33,7 @@ end
 
   def new(%{filename: filename, binary: binary}) do
     %Arc.File{binary: binary, file_name: Path.basename(filename)}
+    |> write_binary()
```

Comment:
I would suggest keeping `new/1` without side-effects. Although this might be a contrived example, imagine a situation where someone is transferring a very large file over the network and they **are not** using `Plug.Upload` or any related mechanism to do so. Now, they _could_ store the filename and binary data as part of a `GenServer` state, but they might also want to store a copy of `Arc.File` instead, since it already contains the filename and binary data in its structure.

And sure, someone could easily create just use `%Arc.File{...}` to create a structure without side-effects, but I don't think `new/1` would be an obvious place for a side-effect. Also imagine trying to test some code that requires an `Arc.File` struct but **does not** require an actual file to exist, or maybe **can't** write a file for some reason; this would have negative impacts on those test cases where some _other_ flow attempts to use `new/1` and the whole thing breaks even though the app code never directly called `new/1`.